content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md

```diff
@@ -170,6 +170,33 @@ kubelet instance. A new kubelet instance deletes all the existing Unix sockets u
 `/var/lib/kubelet/device-plugins` when it starts. A device plugin can monitor the deletion
 of its Unix socket and re-register itself upon such an event.
 
+### Device plugin and unhealthy devices
+
+There are cases when devices fail or being shut down. Device Plugin responsibility
+in this case is to notify kubelet about this situation using the `ListAndWatchResponse` API.
+
+Once device is marked as unhealthy, kubelet will decrease the allocatable count
+for this resource on the Node to reflect how many devices can be used for scheduling new pods.
+Capacity count for the resource will not change.
+
+Pods that were assigned to the failed devices will continue be assigned to this device.
+It is typical that code relying on the device will start failing and Pod may get
+into Failed phase if `restartPolicy` for the Pod was not `Always` or enter the crash loop
+otherwise.
+
+Before the Kubernetes version 1.31, the way to know whether Pod is associated with the
+failed device was to use the [PodResources API](#monitoring-device-plugin-resources).
+
+{{< feature-state feature_gate_name="ResourceHealthStatus" >}}
+
+By enabling the feature gate `ResourceHealthStatus`, the new field `AllocatedResourcesStatus`
+will be added to each ContainerStatus in the pod status object that will contain
+Health information for each device assigned to the container.
+
+This status can be used to understand whether pod failure or misbehavior was associated
+with the device failure.
```

Comment:
Does that example look correct?

Reply:
(would like confirmation I guessed right)

Reply:
The device plugin API doesn't have provision to report unhealthy reason. The device will thus report to the kubelet just healthy/unhealthy. The current user-facing API (in the pod object) also report just healthy/unhealty. So the example is realistic, but in the pod status the user won't actually find "unhealthy because overheating" in 1.31, but just "unhealthy". @SergeyKanzhelev to keep me honest here.

Reply:
OK cool. Polishing this up post-merge is fine.

How do we feel about a separate controller annotating the Pod with more detail vs. reporting an Event vs. neither of those?

Reply:
@sftim IMO this is great input for the next stages of this work!